project.md

```diff
@@ -5,47 +5,168 @@
+quality isn't the greatest nor can we trust that the performers themselves did
+it right. Additionally, we can't really google any of these because again, these
+songs are passed down from instructor to student orally.
+
+Just the other week, we had a minor incident where no one could agree how
+exactly a song went. We ended up deciding on something, but it's very possible
+that we were not correct and therefore altered how the song originally
+went. Thus, I think it would be very helpful to have some kind of physical aid
+to remember how a song goes instead of relying on faulty memory.
+
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+Taiko songs into sheet music.
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+I want to make it somewhat similar to MuseScore so that people don't have to
+learn entirely new controls. You would be presented with a GUI where you have a
+sheet of empty staffs in front of you, and then you could click what notes you
+want to put on it. The notes would be colored/shaped differently depending on
+what hand you should play with and where you should hit the drum.
+
+To make it more convenient, I could have a library of common rhythms (such as
+the horse beat, which is a bit of a swing that sounds like a horse galloping)
+that people could click to drop a collection of notes onto a measure, rather
+than doing it one by one. 
+
+Again, I believe this is the right way because musicians are already familiar
+with GUIs like MuseScore and it clearly works well; there isn't really a need to
+break the formula here except to introduce something that doesn't exist yet (the
+taiko sheet music).
+
+Another thought I have in mind, though, is a simple text input where you input
+the rhythm (for example, writing don don don don would create 4 quarter notes)
+and run the program to make it show up on sheet music. This might be more
+convenient for taiko players specifically because that is how they are taught
+the music. 
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When the program runs, it should export the sheet music as a PDF that can then
+be printed. The user can then read off of the sheet music. Another functionality
+that might be explored in the future is playing back what the music would sound
+like from the program directly. 
+
+Some errors might be trying to fit more notes than there is space (in this case,
+we just deny the user and prevent them from clicking until they delete notes to
+make space). 
+
+Because music is very free-form, though, I don't think there's a lot of errors
+they could make aside from going beyond the constraints of the page.
```

Comment:
People could try and use the GUI in weird ways, like overwriting existing notes with new ones. And errors there would need to be handled delicately (with graphical pop-ups instead of command-line messages). Something to consider as the GUI comes together!